src/Storage/Storage.Management/StorageAccount/RestoreAzStorageBlobRange.cs

```diff
@@ -119,47 +122,95 @@ public override void ExecuteCmdlet()
                     try
                     {
-                        waitTask.Wait();
+                        var result = restoreLro.WaitForCompletion().Value;
+                        WriteObject(new PSBlobRestoreStatus(result));
                     }
                     catch (System.AggregateException ex) when (ex.InnerException is CloudException)
                     {
-                        throw new InvalidJobStateException(string.Format("Blob ranges restore failed with information: '{0}'.", ((CloudException)ex.InnerException).Response.Content));
+                        throw new InvalidJobStateException(string.Format("Blob ranges restore failed with information: '{0}'.", ex.ToString()));
                     }
-
-                    AzureOperationResponse<BlobRestoreStatus> result = waitTask.Result;
-
-                    WriteObject(new PSBlobRestoreStatus(result.Body));
-
                 }
                 else
                 {
-                    BlobRestoreStatus status = this.StorageClient.StorageAccounts.BeginRestoreBlobRanges(
-                     this.ResourceGroupName,
-                     this.StorageAccountName,
-                     this.TimeToRestore,
-                     PSBlobRestoreRange.ParseBlobRestoreRanges(this.BlobRestoreRange));
-
-                    WriteObject(new PSBlobRestoreStatus(status));
-                    if (status != null && status.Status == BlobRestoreProgressStatus.Failed)
+                    Track2.StorageAccountResource account = this.StorageClientTrack2.GetStorageAccount(this.ResourceGroupName, this.StorageAccountName);
+                    var restoreLro = account.RestoreBlobRanges(
+                        WaitUntil.Started,
+                        new Track2Models.BlobRestoreContent(
+                            this.TimeToRestore.ToUniversalTime(),
+                            PSBlobRestoreRange.ParseBlobRestoreRanges(this.BlobRestoreRange)));
+                    Dictionary<string, object> temp = restoreLro.GetRawResponse().Content.ToObjectFromJson() as Dictionary<string, object>;
+                    object jobStatus = null;
+
+                    if (temp != null && temp.TryGetValue("status", out jobStatus))
+                    {
+                        if (jobStatus != null && jobStatus.ToString() == Track2Models.BlobRestoreProgressStatus.Failed.ToString())
+                        {
+                            throw new InvalidJobStateException("Blob ranges restore failed.");
+                        }
+                    } else
                     {
-                        throw new InvalidJobStateException("Blob ranges restore failed.");
+                        WriteWarning(string.Format("Could not fetch the status."));
                     }
+
+                    temp.TryGetValue("restoreId", out object restoreId);
```

Comment:
`temp.***` is not guarded by the null-check.

Reply:
Changed to throw an exception when temp is null